src/Database/Esqueleto/Experimental.hs

```diff
@@ -440,26 +654,56 @@ data From a where
 -- \`on\` (\\(p :& bP) ->
 --         p ^. PersonId ==. bP ^. BlogPostAuthorId)
 -- @
---
-on :: ToFrom a => a -> (b -> SqlExpr (Value Bool)) -> (a, b -> SqlExpr (Value Bool))
+class ValidOnClauseValue a where
```

Comment:
It seems like this is just a replacement for `ToFrom` - can you make a note on what it's all about? Should an end user ever be defining this?